pkgs/by-name/me/meson/package.nix

```diff
@@ -90,15 +90,10 @@ python3.pkgs.buildPythonApplication rec {
     # Fix extraframework lookup on case-sensitive APFS.
     # https://github.com/mesonbuild/meson/pull/13038
     ./007-case-sensitive-fs.patch
-
+  ] ++ lib.optional (stdenv.cc.isZig)
     # Fix meson's detection for zig's linker
     # https://github.com/mesonbuild/meson/pull/12293
-    (fetchpatch {
-      name = "linker-support-zig-cc.patch";
-      url = "https://github.com/mesonbuild/meson/pull/12293/commits/2baae244c995794d9addfe6ed924dfa72f01be82.patch";
-      hash = "sha256-dDOmSRBKl/gs7I3kmLXIyQk3zsOdlaYov72pPSel4+I=";
-    })
-  ];
+    ./linker-support-zig-cc.patch;
```

Comment:
Why are we patching this in Nixpkgs at all instead of fixing it properly upstream?  My read of the upstream PR was that upstream was responsive and interested, and that there was just the small issue of getting the test suite fixed left to do, and upstream had even provided a hint for how to do that.

Reply:
Because there's already an upstream PR. This is just a vendored copy after reading the feedback left after #318034 was merged. I don't see anything about a test suite fix and I'm not familiar with how meson works.

Reply:
I understand that this is just vendoring the patch, but I'm not thrilled that it was added in the first place ­— the original author of the upstream PR doesn't seem to be continuing with it, which means that we have to keep this patch around (with the implied maintenance burden on Nixpkgs' Meson maintainers) until somebody else picks it up.  We can add patches to Nixpkgs when they're urgent, when it's not feasible to upstream them, or when it's doing something Nixpkgs-specific, but allowing patches like this, where none of those things are true, means that, for non-maintainers, there's never any incentive to upstream anything, resulting in packages getting more and more complicated until they collapse under their own weight.  For that reason, I think this patch needs to be sorted out upstream, and that we should not keep it around in Nixpkgs in the meantime.  The path to getting it upstream is clear, and doesn't sound like it would be a lot of work.

> I don't see anything about a test suite fix

https://github.com/mesonbuild/meson/pull/12293#issuecomment-1920147570

Reply:
Yeah, @tomberek and I talked yesterday about this patch. I only added it so meson projects would at least configure in `pkgsZig`. I understand the patch isn't exactly stable but there's not really any other way around the error which occurs without this patch. The problem looks like tests in meson fail because Zig doesn't support a flag which means we likely would have to implement the `-rpath-link` flag.

Reply:
> there's not really any other way around the error which occurs without this patch.

Well, there is, and it's to help this get fixed upstream.

Reply:
Alright, I've made a patch https://github.com/mesonbuild/meson/pull/13493

Reply:
Great, thanks.  For testing on NixOS I just use a nix-shell with python3 and whatever else the test suite needs.

Reply:
Yeah it looks like I'll need to pull in the CC wrapper for the tests to perform correctly.